network/network.go

```diff
@@ -1073,6 +1073,9 @@ func (n *network) dial(ctx context.Context, nodeID ids.NodeID, ip *trackedIP) {
 			timer := time.NewTimer(ip.getDelay())
 
 			select {
+			case <-ctx.Done():
+				timer.Stop()
+				return
```

Comment:
Should we be treating this the same way that we treat unwanted connections? i.e. should we be doing:
```
n.peersLock.Lock()
if ip, exists := n.trackedIPs[nodeID]; exists {
	ip.stopTracking()
	delete(n.peerIPs, nodeID)
	delete(n.trackedIPs, nodeID)
}
n.peersLock.Unlock()
```
here?

Reply:
Actually maybe we would want something like:
```
n.peersLock.Lock()
if trackedIP, exists := n.trackedIPs[nodeID]; exists && ip == trackedIP {
	ip.stopTracking()
	delete(n.peerIPs, nodeID)
	delete(n.trackedIPs, nodeID)
}
n.peersLock.Unlock()
```

Reply:
I thought this would be okay to not include because the context is cancelled on shutdown and we clear the tracked ips list during shutdown

Reply:
Talked with dan offline and we decided to remove the context from the function arg to enforce that it is always the shutdown context